### ISSUE TITLE
Redesign + refactoring

### ISSUE THREAD

**Issue by katrin-k** [rubycorns/rorganize.it#468]
@lislis, @sareg0 and me talked about how to implement the new design. We'll have a new directory under http://rorganize.it/redesign. 

There we can work on 
- refactoring scss files
- implement the new layout
- work on UX issues (until a certain point that needs tbd)

At some point, launch all this

**Comment by sareg0** [comment]
@katrin-k when you make your changes to your files and they are deployed, they will be viewable via https://rorganize.it/?redesign=true thanks to @BriocheBerlin and her brilliant PR: 
https://github.com/rubycorns/rorganize.it/pull/471

Here is how your workflow should look.

Use [application.css.scss](https://github.com/rubycorns/rorganize.it/blob/master/app/assets/stylesheets/application.css.scss) as a guide. In this file we `require` the files/folder we are going to use in the app.
Include all of your files/folder in the new [redesign.css.scss file](https://github.com/rubycorns/rorganize.it/blob/master/app/assets/stylesheets/redesign.css.scss)

Put these in a folder called `redesign` or something of the like (if you haven't already), and place this folder inside the `[app/assets/stylesheets` folder](https://github.com/rubycorns/rorganize.it/tree/master/app/assets/stylesheets).

You can push your changes and make pull requests, and as long as you have required the appropriate files and folders in `redesign.css.scss`, you will see them live via https://rorganize.it/?redesign=true

Do ping us as you go, and we can help you if you want to pair/need advice/etc.

**Comment by til** [comment]
Lots of changes already in the redesign branch, very cool! I'm very curious to see how it is going to look!

It would be great if we can make all those changes previewable to us and others via https://rorganize.it/?redesign=true (right not this looks still a bit, uhm, basic). For that to work, two parts are needed: first, the switching of the css files, which is already accomplished in the master branch (also thanks to @sareg0's detailed instructions!) Second, a way to have two different version of templates (`.html.haml`) files in parallel, since those change as well in the redesign.

We could use a feature of rails called Action Pack Variants for this. http://edgeguides.rubyonrails.org/4_1_release_notes.html#action-pack-variants . Basically it would allow to add a redesigned variant of a template next to the original version, like this:

```
app/views/coaches/index.html.haml
app/views/coaches/index.html+redesign.haml
```

and Rails would automatically use the redesign variant when called with `redesign=true`. What do you think about this - anyone wants to give it a try? (I'll be at Asquera tomorrow and happy to help in person).

**Comment by BriocheBerlin** [comment]
Ok, I'll give it a try.

**Comment by til** [comment]
The preview is live: https://rorganize.it/?redesign=true

(after clicking on a link, the redesign param is gone and needs to be added manually. Anybody have an idea if that is easily fixable?)

**Comment by katrin-k** [comment]
Found a way appending the `?redesign=true` param to each and every href on a page with JS if the initial URL contains the param. 
```
var url = window.location;
var urlString = url.toString();
var substring = 'redesign';

function appendRedesignParam() {
	if (urlString.indexOf(substring) !== -1) {
		$('a').each(function(){
		 this.href += '?redesign=true';
		})
	}
}
$(document).on('page:load', appendRedesignParam);
```
and then triggering the function from inside the JQuery.ready(function…) block. 

Works:
1) Load page https://rorganize.it/?redesign=true
2) Happily browse through the new design

PR will follow.

**Comment by til** [comment]
That's a great approach, looking forward to the PR!

**Comment by katrin-k** [comment]
I sketched a couple versions for a new groups#show page. I feel, both have their pro's and con's. What do you all think? (Click opens .png)

The "tabbed content" follows @lislis' sketch. Focus is on people and contact.
[Tabbed content](https://cloud.githubusercontent.com/assets/2246045/17677099/90659d40-6331-11e6-8482-6d69ef7d5a82.png)

This approach tries to be a complete overhaul. Focus is on group's content.
[Contact in right sidebar](https://cloud.githubusercontent.com/assets/2246045/17677097/90439df8-6331-11e6-9b4c-df69897be033.png)

**Comment by til** [comment]
I like both versions, but would slightly prefer the one focusing on the
group's content. What do the others think, anyone has a strong opinion?

I would suggest to implement the redesign of the template as much as
possible with the current capabilities of the app, and postpone
functionalities that it doesn't currently have, such as e.g. the
"Beginner" label, in order to get the redesign as soon as possible in a
state that we can put it online (oh that would be awesome, wouldn't
it?).

**Comment by PragTob** [comment]
Agree with Til. Minus that I slightly prefer the first sketch (looks cleaner to me).

**Comment by lislis** [comment]
I also prefer the first sketch (lol ;)) I think our main use case it that people try to find a group and see who is already a member. So, I think that makes the most sense? 
But if we orientate ourselves to be more 'organising the group life' then the second one would be better suited.

**Comment by til** [comment]
I would have thought that the main use case is someone trying to find a
group, not knowing anyone in any group, and judging by how the group
describes itself, what curriculum they offer, what topics they plan to
cover etc.

I'd say a matter of taste ;-), both sketches look cool and support both
use cases sufficiently.

**Comment by katrin-k** [comment]
Thanks for your feedback and ideas. So I think the way to go will be the first sketch, but with another ordering of the tabs (first the content, members are last).

**Comment by katrin-k** [comment]
Quite some good news: Due to an unexpected occurrence of insomnia I've been able to finish rewriting + restyling all the pages. PR is in. 

There is still something to do (cc @lislis )
- [ ] Test the hell out of the page. 
- [ ] Button pattern (somewhat consistent atm, but not really. Another button style might also be in order.)
- [ ] All links need care. They currently don't match the design by Justine. 
- [ ] Add media queries where necessary
- [ ] Enhance form styling for error highlighting, focus states, etc.
- [ ] Check all files for "TODO"-comments: Here I mentioned when I had a problem somewhere (e.g. tabbed content on groups#show, but there are plenty more)
- [ ] Check all files for missing accessibility markup

**Comment by sareg0** [comment]
Heya @katrin-k how would you like us to report things/help with this? Would you all things we spot to be reported on this issue? Or would you prefer to pair on it?

**Comment by katrin-k** [comment]
puuh… I'm not sure what makes the most sense. Report issues here or somewhere else… Here is probably fine. 

And yes, it'd be wonderful if you could all browse through the new layout and list errors, things that look strange, maybe even ideas that might be separate issues.

**Comment by katrin-k** [comment]
I'll be closing this issue since relaunch ✨
